### PR TITLE
Fix pagination and add count of conversations

### PR DESCRIPTION
**Pull request by dcdarrell9**
**What is the context of this PR?**

Added the functionality for pagination plus refactored 'count' to be able to count all messages first then count any messages when a label gets passed through ('unread' etc.)

**How to review**

Needs to be run alongside 'SM127-internal-pagination' in [response-operations-ui](https://github.com/ONSdigital/response-operations-ui/pull/130).
Check code and run tests.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "python.pythonPath": "/usr/local/bin/python3",
+    "python.linting.pylintEnabled": true,
+    "python.linting.enabled": true
+}
```
Comment (AdamHawtin) [inline .vscode/settings.json]: Don't think this vscode file was meant to be committed.

File: secure_message/v2/repository/retriever.py [hunk 2]
```diff
@@ -33,6 +33,24 @@ def unread_message_count_by_survey(user, survey):
...
+
+        try:
+            result = SecureMessage.query.join(Status).\
+                filter(or_(*status_conditions)).\
+                filter(and_(*survey_conditions)).\
+                distinct(SecureMessage.thread_id).count()
+        except Exception as e:
+            logger.error('Error retrieving count of messages by survey from database', error=e)
+            raise InternalServerError(description="Error retrieving count of unread messages from database")
+        return result
```
Comment (ajmaddaford) [inline secure_message/v2/repository/retriever.py]: This method shares a lot of code with `unread_message_count_by_survey`, can we rationalise into one method where label is a parameter?

File: secure_message/v2/resources/messages.py [hunk 3]
```diff
@@ -39,15 +39,16 @@ class MessageCounterV2(Resource):
...
-                return jsonify(name=name, total=message_service.unread_message_count_by_survey(g.user, survey))
+                return jsonify(name=label, total=message_service.message_count_by_survey(g.user, survey, label))
             else:
-                logger.debug('Invalid label name', name=name, request=request.url)
+                logger.debug('Invalid label name', name=label, request=request.url)
                 raise BadRequest(description="Invalid label")
         else:
-            logger.debug('No Name parameter specified in URL', request=request.url)
-            raise BadRequest(description='No Label Name Parameter specified.')
+            label = None
```
Comment (ajmaddaford) [inline secure_message/v2/resources/messages.py]: If you default the label parameter in `message_count_by_survey` to `None`, this variable isn't needed.

File: secure_message/v2/repository/retriever.py [hunk 4]
```diff
@@ -16,20 +16,31 @@ class RetrieverV2(Retriever):
...
-    def unread_message_count_by_survey(user, survey):
-        """Count users unread messages for a specific survey"""
+    def message_count_by_survey(user, survey, label):
+        """Count users messages for a specific survey"""
         if user.is_internal:
             status_conditions, survey_conditions = RetrieverV2._get_conditions_internal_user(survey, user)
         else:
             status_conditions, survey_conditions = RetrieverV2._get_conditions_respondent(survey, user)
 
+        if label is 'UNREAD':
```
Comment (ajmaddaford) [inline secure_message/v2/repository/retriever.py]: You could drop the `UNREAD` here and just do:
```Python
if label:
    ...
    filter(Status.label == label).count()
```

**Comment by dcdarrell9** [comment]
[SM127: INT Pagination (8)](https://trello.com/c/VqG2E8Jm/116-sm127-int-pagination-8)

**Pull request merged**